src/persist/src/operators/stream.rs

```diff
@@ -303,6 +308,57 @@ where
     }
 }
 
+/// Extension trait for [`Stream`].
+pub trait AwaitFrontier<G: Scope<Timestamp = u64>, D> {
+    /// Stashes data until it is no longer beyond the input frontier.
+    ///
+    /// This is similar, in spirit, to what `consolidate()` does for differential collections and
+    /// what `delay()` does for timely streams. However, we are not working with differential
+    /// collections and `delay()` deals with changing the timestamp while the behaviour we want is to
+    /// wait for the frontier to pass. The latter is an implementation detail of `delay()` that is not
+    /// advertised in its documentation.
+    fn await_frontier(&self, name: &str) -> Stream<G, (D, u64, isize)>;
+}
+
+impl<G, D> AwaitFrontier<G, D> for Stream<G, (D, u64, isize)>
+where
+    G: Scope<Timestamp = u64>,
+    D: TimelyData,
+{
+    // Note: This is mostly a copy of the timely delay() operator without the delaying part.
+    fn await_frontier(&self, name: &str) -> Stream<G, (D, u64, isize)> {
+        let operator_name = format!("await_frontier({})", name);
+
+        let mut elements = HashMap::new();
```

Comment:
If I'm reading this right this HashMap is a `HashMap<Timestamp, Vec<Vec<(D, u64, isize)>>>` right? I'm not sure but would it be worth it to only have a single vector, instead of a vec of vecs? I know timely does some strange RefOrMut stuff so I'm not sure. My main reason for asking is less performance and more just simplicity and ease of understanding what's going on.